client/src/components/ProtectedRoutes/ProtectedRoutes.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Redirect, Route } from 'react-router-dom';
+
+import { useAuth } from '../../context/useAuthContext';
+
+interface RouteProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  component: any;
+  path?: string;
+  exact?: boolean;
+}
+
+export default function ProtectedRoute({ component: Component, ...rest }: RouteProps): JSX.Element {
```

Comment:
In the RouteProps you specify component is `any` but here you declare it's type.